features/2025-07-10-offline-install-feature-spec.md

```diff
@@ -0,0 +1,222 @@
+
+**Radius and AWS Bicep extensions**
+
+The platform engineer or security engineer imports the Radius Bicep extensions into their OCI registry. 
+
+### User Story 2 – Install CLI
+
+***As a platform engineer or developer, I need to install the Radius CLI using pre-downloaded binaries***
+
+The platform engineer or developer runs the installation script with additional parameters.
+
+```bash
+$ install.sh \
+  --version <radius_version>
+  --binary-install-source <path_to_rad_and_rad_bicep_cli> \
+  --bicep-extension-oci-registry <oci_registry_url>/<repository>
+```
+
+This script performs the following:
+
+1. Gets the kernel type and system architecture (as-is)
+2. Downloads the `rad` binary from Github (as-is) or the binary install source (new)
+3. Gets the latest release information if the version is not specified (as-is)
+4. Installs the `rad` binary in `/usr/local/bin` or `%LOCALAPPDATA%\radius` (as-is)
+5. Installs the `rad-bicep` binary in `/usr/local/bin` or `%LOCALAPPDATA%\radius` (change from as-is)
+6. Creates the `bicepconfig.json` in the user's home directory (new) using the `bicep-extension-oci-registry`, if specified (new)
+
+The key changes from today's `install.sh` include:
+
+* Today, `rad-bicep` is installed via the `rad bicep download` command. This approach is problematic in that is is not obvious that Radius is downloading a new binary. The command also installs the `rad-bicep` binary in a different directory (`~/.rad/bin`) which is not ideal. The `install.sh` script is enhanced to install the `rad-bicep` binary side-by-side `rad` and the `rad bicep download` command is removed.
+* Today, the `bicepconfig.json` file is created by `rad init`. This is problematic since offline installs will not use `rad init`. Also, the contents of `bicepconfig.json` file created by `rad init` is hard-coded. With this change, the `install.sh` script will create the `bicepconfig.json` file based on user input (`--bicep-extension-oci-registry`).
+
+### User Story 3 – Install Radius via CLI
+
+***As a platform engineer, I need to install Radius offline using the Radius CLI***
+
+The platform engineer installs Radius using the CLI specifying the private OCI registry and to skip installing Contour.
+
+```bash
+rad install kubernetes \
+  --set global.imageRegistry=<oci_registry_url>
+  --set global.imageTag=<image_tag>
+  --set global.imagePullSecrets=<kubernetes_secret_name>
+  --skip-contour-install
+```
+
+> [!NOTE]
+>
+>  `rad init` is out of scope for offline installations. We expect `rad init` to continue to be optimized for the quick start scenario. 
+
+The `global.imageRegistry` is the private OCI registry where the Radius container images were stored in the previous step.
+
+The `global.imageTag` is the tag of the container images to optionally be used.
+
+The `global.imagePullSecrets` is the name of the Kubernetes secret of type docker-registry to be used to pull the container images (the secret is created beforehand).
```

Comment:
This is an array: https://github.com/radius-project/radius/blob/main/pkg/cli/cmd/install/kubernetes/kubernetes.go#L76.